splice_machine/src/main/java/com/splicemachine/derby/impl/storage/CheckTableJob.java

```diff
@@ -299,19 +303,63 @@ private ExecRow getIndexExecRow(long conglomerateId) throws StandardException {
                         .keyDecodingMap(ScanOperation.getKeyDecodingMap(accessedKeyColumns, columnOrdering, baseColumnMap))
                         .rowDecodingMap(ScanOperation.getRowDecodingMap(accessedKeyColumns, columnOrdering, baseColumnMap))
                         .baseColumnMap(baseColumnMap)
+                        .defaultRow(defaultValueRow, defaultValueMap)
                         .buildDataSet();
 
-        //PairDataSet<String, byte[]> dataSet = scanSet.index(new KeyByRowIdFunction());
-
         return scanSet;
     }
 
+    private FormatableBitSet getDefaultValueMap(ExecRow defaultValueRow) throws StandardException {
+        FormatableBitSet defaultValueMap = null;
+        int size = 0;
+        for (int i = 0; i < defaultValueRow.nColumns(); ++i) {
+            if (!defaultValueRow.getColumn(i+1).isNull())
+                size++;
+        }
+
+        int index = 0;
+        if (size > 0) {
+            defaultValueMap = new FormatableBitSet(size);
```

Comment:
I think the size of FormatableBitSet should be the same as the number of columns in defaultValueRow, and each bit that is set represent that the corresponding column's default value may be used to fill the row, however, the defaultMap set here does not seem so.